### DISCUSSION TITLE
Add option to set `request_methods_allowed_content`?

### DISCUSSION THREAD

**Discussion by MSP-Greg** [puma/puma#3554] [Ideas]
Currently, Puma allows request content (bodies) with any request method.  RFC 9110 indicates that servers may reject (close) requests that include content where the content has no meaning or effect on the server’s actions and/or response.

Should an option similar to `request_methods_allowed_content` be available, which is used to define the request methods that are allowed content?  The option’s value would be an array of methods.

**Comment by dentarg** [comment]
Is it a common feature in other web servers?

**Reply by MSP-Greg** [reply]
I've been meaning to check nginx, but haven't yet.  I should probably also check a node.js server, but I've never looked into that...
